### ISSUE TITLE
Doctest failure: tutorial/type_check.rst

### ISSUE THREAD

**Issue by niboshi** [chainer/chainer#3273]
Current master (0036b0421d18d38dceefd6a866e741bc0d2da5f3)

`make doctest` log
```
Warning, treated as error:
WARNING: **********************************************************************
File "tutorial/type_check.rst", line 117, in default
Failed example:
    utils.type_check.expect(x_type.ndim == 2)
Expected:
    Traceback (most recent call last):
    ...
    InvalidType: Expect: in_types[0].ndim == 2
    Actual: 3 != 2
Got:
    Traceback (most recent call last):
      File "/anaconda/anaconda3/lib/python3.6/doctest.py", line 1330, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        utils.type_check.expect(x_type.ndim == 2)
      File "/repos/chainer/chainer/utils/type_check.py", line 514, in expect
        expr.expect()
      File "/repos/chainer/chainer/utils/type_check.py", line 472, in expect
        '{0} {1} {2}'.format(left, self.inv, right))
    chainer.utils.type_check.InvalidType: Expect: in_types[0].ndim == 2
    Actual: 3 != 2


Makefile:176: recipe for target 'doctest' failed
make: *** [doctest] Error 1
```

Environment
```
Python 3.6.1 :: Anaconda 4.4.0 (64-bit)
Sphinx (sphinx-build) 1.5.6
```

**Comment by unnonouno** [comment]
It is caused by python3 because format of Exception is different between py2 and py3.

**Comment by niboshi** [comment]
OK, let's keep this issue open.

**Comment by niboshi** [comment]
It seems this has been fixed in #3653?
